tests/tape/tapes/test_qnode.py

```diff
@@ -757,3 +757,86 @@ def circuit(inputs, weights):
                 qnode(x, p, parallel=True)
         except:
             pytest.fail("Multi-threading on QuantumTape failed")
+
+
+class TestIntegration:
+    """Integration tests."""
+
+    def test_correct_number_of_executions_autograd(self):
+        """Test that number of executions are tracked in the autograd interface."""
+        qml.enable_tape()
+
+        def func():
+            qml.Hadamard(wires=0)
+            qml.CNOT(wires=[0, 1])
+            return qml.expval(qml.PauliZ(0))
+
+        dev = qml.device("default.qubit", wires=2)
+        qn = QNode(func, dev, interface="autograd")

```

Comment:
Should we add `diff_method="backprop"` here to be explicit, in case we ever change the logic?

Reply:
But if we change the logic and don't switch the device any more, we actually don't only want to test "backprop" either. Could do a fixture with all diff methods?